lib/srv/ctx.go

```diff
@@ -722,6 +726,15 @@ func (c *ServerContext) ExecCommand() (*ExecCommand, error) {
 		requestType = c.request.Type
 	}
 
+	// This block of code gathers and prepares metadata thats needed for user accounting.
```

Comment:
if you moved this to a function called `collectMetadataForUacc`, the comment would not be needed.
It's also a good idea to keep this in an individual struct.

Reply:
Moved this stuff into a seperata struct and collection function.